mmv1/third_party/terraform/services/container/resource_container_node_pool_test.go.erb

```diff
@@ -1644,6 +1644,64 @@ resource "google_container_node_pool" "np" {
 `, cluster, networkName, subnetworkName, np)
 }
 
+func TestAccContainerNodePool_secondaryBootDisks(t *testing.T) {
```

Comment:
Looks like this testcase is failing

```
2024-04-23T23:11:37.685Z [DEBUG] sdk.helper_resource: Stopping providers: test_step_number=1 test_name=TestAccContainerNodePool_secondaryBootDisks test_terraform_path=/bin/terraform test_working_directory=/tmp/plugintest2933754998
2024-04-23T23:11:37.686Z [ERROR] sdk.helper_resource: Unexpected error: test_working_directory=/tmp/plugintest2933754998 test_step_number=1
  error=
  | Error running apply: exit status 1
  | 
  | Error: NodePool tf-test-nodepool-a0e4o4evy9 was created in the error state "ERROR"
  | 
  |   with google_container_node_pool.np,
  |   on terraform_plugin_test.tf line 12, in resource "google_container_node_pool" "np":
  |   12: resource "google_container_node_pool" "np" {
  | 
```

Reply:
[debug link](https://ffdafa6f9d56f6863f9a729be30995dc5fd02976c605b09a56394d7-apidata.googleusercontent.com/download/storage/v1/b/ci-vcr-logs/o/beta%2Frefs%2Fheads%2Fauto-pr-10511%2Fartifacts%2Fe8d2577e-c2dd-40da-9be6-c25cb34cde98%2Frecording%2FTestAccContainerNodePool_secondaryBootDisks.log?jk=ASOpP9huGA7xScySgGeufaNc5273EucD9d1FAtHK3HNvC8cRg8MSdjix2da5eZcOL7BhtJAXqVvrELDkC5fvubJWhJXmWRVTrABK-NCRcmBbKNJlJVa_mxC-JsdKLdatm2OMjrFsvIHYkshezdtdVz7NpDTeKGanAmBMpasErrJWbrj09e5vCZ2UxJB8DfTnNPsaw_vdQzuaA9TA4Xrk60hUnZjXyIf5UDPmcs5ayGW7mWJEep6f8e8hpg5a5PwNt9svxnV6oWeHAk3oqFJc8wlAh2rOOGG8ByrdxDqFKbRXLmEo5KlHqbwW6vzGst-r4qe2L4iQMkdHfbzsr-TVpRvcIaME2uX7h3p92AP2R7vcjPeKU0Asve87nP-IFPIyEa-Ajjh49OSUzAlAsOlLKV9dSHzt4c1sxFzYQJI9rqC62HPENIPAORWc8FrjklNrjJsyUmMEhtGIs9yyHHQ7zAGvOl_-tfAb5SmKLQvAdITlNpMl5pI_zxV8VorUO24bo4P1sXlJ7Yb4XCXu8enqYEWYxLYFa7-zEvMACvKk2eZM6aV5W0H7MKdHImwbeswiWYFCaryJiDl6xQCGLOIXM75jNcZqLgVWNd82nFDLMYxL12WN73jn9B6LcNiBYpkOTkRTBEzHtRUxVIIZjWNofZlVHoD7wNIfKD-Ioi466WiDQOGwOTBnomZZ3m_DaSN8hJs8WIbG6TFFrqqTUIH2o7Gl28rLhgkpPBOniDKPJSNxcfB60LF3eCzJ8-L_NBr8eFLgxBoURDc7VwKReFkye_SXMNZoOoqWaRzDK59mwQNtdulvLR3TDEZ1aixG2vVuy_XMKd30Fo27JeFV2Cw1NHF2kFcoklop2dwaUJjo7cEu0-0ke_8Cc85FTtbwcpy8WxprZ8tCsDnpwkHkjrsS1ypJM5ScJhrbzEekNx680QjneGt_hu5YXV4mYtrG61_S-rFYZnBtQEXW--gsqB9udlCmQvPSr3TCrfNh48IpCUAv4h1gtvejcvZXwsSNpmc-16T_BWcILBg9SNMgcLRBlykAtvTl2YmUc7Py7yuhoF9DdyJo3gOdXQAP4aiMXceJaaq-MHrPM_tjMHe5KpchHRRG2aPGlWWRBUAw5gJja5Z2lE5rmEKX0RefDH7qHG1k4UBE-V3_wigQy1yxEWWn0pP8LN4SuuzTS8Uw66aqp1d-34DdsQvT8nKfsinZDmnRTCuiTNKGd_UJ61QX3kqJxloCrrlqmq_EROayoJhrLcddaYE0BGFlYwR75ceOR2Chjp1kzz3bDQSEsRkdabZL_dVol1lJ92km8Dyx18gHVKVW_vIUbfj26W1q3VR4fZSz_j5-aBqc5hY&isca=1)


The disk image is not there. 

2024/04/23 00:35:57 [DEBUG] Retry Transport: Stopping retries, last request was successful
2024/04/23 00:35:57 [DEBUG] Retry Transport: Returning after 1 attempts
2024/04/23 00:35:57 [DEBUG] NodePool "projects/ci-test-project-188019/locations/us-central1-a/clusters/tf-test-cluster-6mbwk8luv0/nodePools/tf-test-nodepool-ftvmywitqd" has error state "ERROR" with message "Google Compute Engine: The resource 'projects/ci-test-project-188019/global/images/example-image' was not found.".

Reply:
Can we use an existing image
https://github.com/modular-magician/terraform-provider-google-beta/blob/2bc478fc9f9befd4bbe3ccc9ece18d7ca430199c/google-beta/services/compute/resource_compute_image_test.go#L271

Reply:
This fetches the family and an image from that family.

https://github.com/modular-magician/terraform-provider-google-beta/blob/2bc478fc9f9befd4bbe3ccc9ece18d7ca430199c/google-beta/services/compute/data_source_google_compute_image.go#L136-L150

Maybe we could wrap it as a utility function.

Reply:
Leaving the disk image unspecified and setting secondary boot disk mode to DATA creates an empty secondary boot disk. This allows us to still test the feature but bypasses image access issues. [image](https://github.com/GoogleCloudPlatform/magic-modules/assets/39973638/d0243721-61a8-4cdb-990d-31a948d12fc9)

Reply:
Is this an intentional pattern? I worry API validation will make these fail in the future.

Reply:
Empty disk is valid use case. But the primary use is to create from disk image.

Reply:
Agreed, if we have an existing image, it is better to use it.

Reply:
I've approved since we have confirmation that this is an okay usage of the resource. Thanks all.